tremor-script/src/ast/deploy/raw.rs

```diff
@@ -0,0 +1,785 @@
+pub struct PipelineDeclRaw<'script> {
+    pub(crate) start: Location,
+    pub(crate) end: Location,
+    pub(crate) id: String,
+    pub(crate) args: WithArgsRaw<'script>,
+    pub(crate) params: Option<DeployWithExprsRaw<'script>>,
+    pub(crate) query: QueryRaw<'script>,
+}
+
+fn arg_spec_resolver<'script, 'registry>(
+    args_spec: WithArgsRaw<'script>,
+    params_spec: &Option<DeployWithExprsRaw<'script>>,
+    helper: &mut Helper<'script, 'registry>,
+) -> Result<(Vec<Cow<'script, str>>, Value<'script>)> {
+    let mut args: HashMap<Cow<str>, Value<'script>> = HashMap::new();
+    let mut spec = vec![];
+
+    let upper_args = helper.consts.args.clone();
+
+    // Process explicitly declared required/optional arguments
+    for arg_spec in args_spec {
+        match arg_spec {
+            ArgumentDeclRaw::Required(name) => {
+                let moo = name.to_string();
+                let moo: Cow<str> = Cow::owned(moo);
+                spec.push(moo.clone());
+            }
+            ArgumentDeclRaw::Optional(name, value) => {
+                let moo = name.to_string();
+                let moo: Cow<str> = Cow::owned(moo);
+                spec.push(moo.clone());
+                let value = value.up(helper)?;
+                let value = stateless_run_expr!(helper, &value);
+                args.insert(moo, value);
+            }
+        }
+    }
+    // We inject the args here so that our `with` clause can override based on
+    // our specificational arguments and their defaults
+    helper.consts.args = Value::Object(Box::new(args.clone()));
+
+    // Process defaulted arguments via `with` expressions
+    if let Some(params_spec) = up_maybe_params(params_spec, helper)? {
+        for (name, value) in params_spec {
+            let moo = Cow::owned(name);
+            if !spec.contains(&moo) {
+                // FIXME TODO consider duplicate argument specification - ok or ko?
+                spec.push(moo.clone());
+            }
+            let up_value = value;
+            args.insert(moo, up_value.clone_static());
+        }
+        // We inject the args here so that our `script` clause can override based on
+        // our `with` specificational argument overrides and their defaults
+        helper.consts.args = Value::Object(Box::new(args.clone()));
+    }
+
+    // NOTE We don't check for `extra args` here - this is deferred to the resolution in `arg_resolver`
+    helper.consts.args = upper_args.clone();
+    Ok((spec, Value::Object(Box::new(args))))
+}
+
+fn arg_resolver<'script, 'registry>(
+    outer: crate::pos::Range,
+    spec: &[Cow<'script, str>],
+    args_as_params: &Value<'script>,
+    params_spec: &Option<DeployWithExprsRaw<'script>>,
+    helper: &mut Helper<'script, 'registry>,
+) -> Result<Value<'script>> {
+    let mut args = args_as_params.clone();
+    // Process defaulted arguments via `with` expressions
+    helper.consts.args = args_as_params.clone();
```

Comment:
same as above, I'm a bit confused as to what happens here with `helper.consts.args`

Reply:
Now this one looks like I was low on coffee or high on distraction!

Reply:
Outdated and the args/with refactoring is more recent. Closing.